lib/Onyx.ts

```diff
@@ -0,0 +1,597 @@
+/* eslint-disable no-continue */
+import * as Logger from './Logger';
+import cache from './OnyxCache';
+import createDeferredTask from './createDeferredTask';
+import * as PerformanceUtils from './PerformanceUtils';
+import Storage from './storage';
+import utils from './utils';
+import DevTools from './DevTools';
+import type {
+    Collection,
+    CollectionKeyBase,
+    ConnectOptions,
+    InitOptions,
+    KeyValueMapping,
+    Mapping,
+    NullableKeyValueMapping,
+    NullishDeep,
+    OnyxCollection,
+    OnyxEntry,
+    OnyxKey,
+    OnyxUpdate,
+    OnyxValue,
+} from './types';
+import OnyxUtils from './OnyxUtils';
+
+// Keeps track of the last connectionID that was used so we can keep incrementing it
+let lastConnectionID = 0;
+
+// Connections can be made before `Onyx.init`. They would wait for this task before resolving
+const deferredInitTask = createDeferredTask();
+
+/** Initialize the store with actions and listening for storage events */
+function init({
+    keys = {},
+    initialKeyStates = {},
+    safeEvictionKeys = [],
+    maxCachedKeysCount = 1000,
+    shouldSyncMultipleInstances = Boolean(global.localStorage),
+    debugSetState = false,
+}: InitOptions): void {
+    if (debugSetState) {
+        PerformanceUtils.setShouldDebugSetState(true);
+    }
+
+    if (maxCachedKeysCount > 0) {
+        cache.setRecentKeysLimit(maxCachedKeysCount);
+    }
+
+    OnyxUtils.initStoreValues(keys, initialKeyStates, safeEvictionKeys);
+
+    // Initialize all of our keys with data provided then give green light to any pending connections
+    Promise.all([OnyxUtils.addAllSafeEvictionKeysToRecentlyAccessedList(), OnyxUtils.initializeWithDefaultKeyStates()]).then(deferredInitTask.resolve);
+
+    if (shouldSyncMultipleInstances) {
+        Storage.keepInstancesSync?.((key, value) => {
+            const prevValue = cache.getValue(key, false);
+            cache.set(key, value);
+            OnyxUtils.keyChanged(key, value, prevValue);
+        });
+    }
+}
+
+/**
+ * Subscribes a react component's state directly to a store key
+ *
+ * @example
+ * const connectionID = Onyx.connect({
+ *     key: ONYXKEYS.SESSION,
+ *     callback: onSessionChange,
+ * });
+ *
+ * @param mapping the mapping information to connect Onyx to the components state
+ * @param mapping.key ONYXKEY to subscribe to
+ * @param [mapping.statePropertyName] the name of the property in the state to connect the data to
+ * @param [mapping.withOnyxInstance] whose setState() method will be called with any changed data
+ *      This is used by React components to connect to Onyx
+ * @param [mapping.callback] a method that will be called with changed data
+ *      This is used by any non-React code to connect to Onyx
+ * @param [mapping.initWithStoredValues] If set to false, then no data will be prefilled into the
+ *  component
+ * @param [mapping.waitForCollectionCallback] If set to true, it will return the entire collection to the callback as a single object
```

Comment:
I don't think I did 🤔 Can you explain what and where I removed?

Reply:
Before it was

```ts
 * @param {Object} mapping the mapping information to connect Onyx to the components state
 * @param {String} mapping.key ONYXKEY to subscribe to
 * @param {String} [mapping.statePropertyName] the name of the property in the state to connect the data to
 * @param {Object} [mapping.withOnyxInstance] whose setState() method will be called with any changed data
 *      This is used by React components to connect to Onyx
 * @param {Function} [mapping.callback] a method that will be called with changed data
 *      This is used by any non-React code to connect to Onyx
 * @param {Boolean} [mapping.initWithStoredValues] If set to false, then no data will be prefilled into the
 *  component
 * @param {Boolean} [mapping.waitForCollectionCallback] If set to true, it will return the entire collection to the callback as a single object
 * @param {Function} [mapping.selector] THIS PARAM IS ONLY USED WITH withOnyx(). If included, this will be used to subscribe to a subset of an Onyx key's data.
 *       The sourceData and withOnyx state are passed to the selector and should return the simplified data. Using this setting on `withOnyx` can have very positive
 *       performance benefits because the component will only re-render when the subset of data changes. Otherwise, any change of data on any property would normally
 *       cause the component to re-render (and that can be expensive from a performance standpoint).
 * @param {String | Number | Boolean | Object} [mapping.initialValue] THIS PARAM IS ONLY USED WITH withOnyx().
 * If included, this will be passed to the component so that something can be rendered while data is being fetched from the DB.
 * Note that it will not cause the component to have the loading prop set to true. |
 * @returns {Number} an ID to use when calling disconnect
 */
function connect(mapping) {
```

Reply:
Thanks for the explanation, you are right. I brought it back!